od-msspe/src/config.rs

```diff
@@ -0,0 +1,177 @@
+pub(crate) use crate::constants::{
+    ANNEALING_TEMP, DEFAULT_NTTHAL_PATH, DEFAULT_PRIMER3_PATH, DELTA_G_THRESHOLD, DNA_CONC,
+    DNTP_CONC, DV_CONC, KMER_SIZE, MAX_ITERATIONS, MAX_MISMATCH_SEGMENTS, MV_CONC, OVERLAP_SIZE,
+    PRIMER_MAX_HAIRPIN_TH, PRIMER_MAX_SELF_ANY_TH, PRIMER_MAX_SELF_END_TH, PRIMER_MAX_TM,
+    PRIMER_MIN_TM, SEARCH_WINDOWS_SIZE, WINDOW_SIZE,
+};
+use std::path::Path;
+
+use clap::Parser;
+
+#[derive(Parser, Debug)]
+#[command(version, about, long_about=None)]
+pub struct Args {
+    #[arg(short, long)]
+    pub input: String,
+
+    #[arg(short, long)]
+    pub output: String,
+
+    #[arg(long, env = "KMER_SIZE", default_value_t = KMER_SIZE)]
+    pub kmer_size: usize,
+
+    #[arg(long, env = "WINDOW_SIZE", default_value_t = WINDOW_SIZE)]
+    pub window_size: usize,
+
+    #[arg(long, env = "OVERLAP_SIZE", default_value_t = OVERLAP_SIZE)]
+    pub overlap_size: usize,
+    #[arg(long, env = "MAX_MISMATCH_SEGMENTS", default_value_t = MAX_MISMATCH_SEGMENTS)]
+    pub max_mismatch_segments: usize,
+    #[arg(long, env = "MAX_ITERATIONS", default_value_t = MAX_ITERATIONS)]
+    pub max_iterations: usize,
+    #[arg(long, env = "SEARCH_WINDOWS_SIZE", default_value_t = SEARCH_WINDOWS_SIZE)]
+    pub search_windows_size: usize,
+    #[arg(long, env = "MV_CONC", default_value_t = MV_CONC)]
+    pub mv_conc: f32,
+    #[arg(long, env = "DV_CONC", default_value_t = DV_CONC)]
+    pub dv_conc: f32,
+    #[arg(long, env = "DNTP_CONC", default_value_t = DNTP_CONC)]
+    pub dntp_conc: f32,
+    #[arg(long, env = "DNA_CONC", default_value_t = DNA_CONC)]
+    pub dna_conc: f32,
+    #[arg(long, env = "ANNEALING_TEMP", default_value_t = ANNEALING_TEMP)]
+    pub annealing_temp: f32,
+    #[arg(long, env = "MIN_TM", default_value_t = PRIMER_MIN_TM)]
+    pub min_tm: f32,
+    #[arg(long, env = "MAX_TM", default_value_t = PRIMER_MAX_TM)]
+    pub max_tm: f32,
+    #[arg(long, env = "MAX_SELF_DIMER_ANY_TM", default_value_t = PRIMER_MAX_SELF_ANY_TH)]
+    pub max_self_dimer_any_tm: f32,
+    #[arg(long, env = "MAX_SELF_DIMER_END_TM", default_value_t = PRIMER_MAX_SELF_END_TH)]
+    pub max_self_dimer_end_tm: f32,
+    #[arg(long, env = "MAX_HAIRPIN_TM", default_value_t = PRIMER_MAX_HAIRPIN_TH)]
+    pub max_hairpin_tm: f32,
+    #[arg(long, env = "DELTA_G_THRESHOLD", default_value_t = DELTA_G_THRESHOLD, help = "Threshold for dG, default is -9000.0 J/mol")]
+    pub delta_g_threshold: f32,
+
+    // logic based config
```

Comment:
change all logic based config to true/false

yes/no is not best practice @noomzopendream